crates/epoch-snark/src/gadgets/epoch_bits.rs

```diff
@@ -184,24 +175,26 @@ mod tests {
             .flatten()
             .collect::<Vec<bool>>();
 
-        // encode each epoch's bytes to LE and pas them to the constraint system
+        let cs = ConstraintSystem::<Fr>::new_ref();
+        // encode each epoch's bytes to LE and pass them to the constraint system
         let first_epoch_bits = bytes_to_bits(&first_bytes, 256);
         let last_epoch_bits = bytes_to_bits(&last_bytes, 256);
         let bits = EpochBits {
             crh_bits: vec![],
             xof_bits: vec![],
-            first_epoch_bits: to_bool(&first_epoch_bits),
-            last_epoch_bits: to_bool(&last_epoch_bits),
+            first_epoch_bits: first_epoch_bits.iter().map(|b| Boolean::new_witness(cs.clone(), || Ok(*b))).collect::<Result<Vec<_>, _>>().unwrap(), 
```

Comment:
I know these are just tests and it doesn't really matter, but can we allocate constants instead?

Reply:
What's the reason to want to allocate constants here? I'm thinking this should be changed to public inputs instead since that's what they'll be in the actual circuit

Reply:
There's no huge reason. It just seems safer to me, since we don't need to think about whether we have all the constraints or anything.

Reply:
Post-refactor using constants here will either cause the code to panic or trigger a different branch (i.e. https://github.com/celo-org/celo-bls-snark-rs/blob/straka/zexe_refactor/crates/bls-gadgets/src/y_to_bit.rs#L92). I think if we want to exercise this it'll be better to make a separate test using constants

Reply:
Got it. I think it's fine to leave it as is then.